Separate workflow to test unsupported package manager platform

mosteo opened the pull request:
In the end I went for a Linux instead of the suggested Windows in #601 because Windows is at present giving us more problems, with sporadic failures with unidentified causes. Also the test is quicker not requiring the CE setup.

Depends on #621 

This PR adds a new config option, `distribution.disable-detection`, that results in an unknown distribution always being reported.

To break circularity problems, loading the configuration no longer requires to load the manifest, only to locate it. This caused one test to err slightly later (as it does not need the manifest contents), which accounts for the test modification.

mosteo committed:
Workflow to test with unsupported package manager

Fabien-Chouteau commented on .github/workflows/ci-unsupported.yml:
```diff
@@ -0,0 +1,51 @@
+        python-version: '3.x'
+
+    - name: Remove identifying file from OS
+      run: sudo rm -f /etc/os-release
```
This could have unexpected side effects.
If you want we can add a config key to disable distrib features.

mosteo replied:
I will do this. It may be interesting even for normal users to disable using crates not coming from sources.

mosteo committed:
Add config option to disable distro detection

With test

mosteo committed:
Fixes for circular elaboration

mosteo committed:
Fix test for slight change in behavior

Before the last changes to break circularities, we tried to load the manifest
as part of the config loading process. Now we are satisfied with finding a
properly named file at the expected place. This causes one test to not detect
invalid TOML, since the manifest contents aren't really needed for the test.

stcarrez commented:
It seems that the script which checks for crate validity fails to recognize the following `.prepend` environment lines:

```
[environment]
DYNAMO_SEARCH_PATH.prepend = "${CRATE_ROOT}/awa/plugins"
DYNAMO_UML_PATH.prepend = "${CRATE_ROOT}/awa/uml"
```

mosteo commented:
It's a known problem with the javascript TOML library which chokes on some syntax for multilevel tables.
